rfd/0001-tooling-requirements.md

```diff
@@ -0,0 +1,150 @@
+1. Extract the new version from the CHANGELOG.
+2. Create a new Git tag targeting the commit that modified the CHANGELOG, with name `v<new version>-<project name>` with all separator characters (e.g. spaces and underscores) separated by dashes. This tag should be parsable by [this npm package](https://www.npmjs.com/package/semver) that Renovate uses for semver coercion.
+3. Build a new release of the tool, following the general logic of the CD workflow. Either the release version or Git tag may be used where appropriate.
+4. Create a new release, named after the Git tag, that includes the changes from the CHANGELOG. The release should also either include the built artifacts, or instructions on retrieving them (such as `docker pull ghcr.io/gravitational/shared-workflows/some-project:v1.2.3`).
+
+### Project releases
+The process for releasing a new version of a project should be as follows:
+1. PRs are filed to add/modify the project.
+2. The project's CI workflow triggers to validate the changes, codeowners approve, and PRs are merged.
+3. Artifacts are built for each PR to facilitate release testing.
+4. A new PR is filed and merged that updates the changelog to a new version.
+5. The release workflow is triggered, which creates a new tag and publishes a fresh build.
+
+### Dependency management
+Historically this repo has used Dependabot to manage dependency updates. Unfortunately however, Dependabot does not support two key "languages" used by Github Actions: actions themselves (used when building composite actions), and Dockerfiles (used when building Dockerfile actions). As a result, [Renovate](https://docs.renovatebot.com/) will be used for new and migrated projects instead. Renovate is highly flexible and can be configured on a per-project basis.
+
+Renovate will self-hosted and configured to run every 12 hours. Updates for each project should be grouped as much as reasonably possible to reduce the number of PRs opened at any given time. Update PRs will be assigned to code owners for each project. Digest pinning will be used wherever supported.
+
+### Security considerations
+The primary source of security concern with anything living in this repo is that the repo is public. The impact of making a mistake with one of our workflows (such as exposing a sensitive value) is much greater than if the repo was private/internal.
+
+I took a look at the forks, stars, watchers, issues, and PRs for this repo. As far as I can tell, all but two or three of the interactions (stars and watchers) from non-Gravitational Github accounts are from bots. Based on this I think it is reasonable to assume that nobody outside of Gravitational uses this repo. Additionally, I don't personally believe that the few existing tools in this repo have a large value to the open source community. With this information in mind, I believe that we make this repo private. Doing so would alleviate most of the security concerns associated with this RFD.
```

Comment:
I believe `shared-workflows` was originally private, and was made public to work with `teleport` and other repos.  See https://github.com/gravitational/github-terraform/pull/240

According to the docs, I don't think we _can_ make this private:

> Actions and reusable workflows stored in private repositories cannot be used in public repositories.

https://docs.github.com/en/actions/creating-actions/sharing-actions-and-workflows-from-your-private-repository#about-github-actions-access-to-private-repositories

As such, we may consider a different (new?) repo.  I don't think `teleport.e` is the best fit for these tools (getting bogged down in teleport CI sucks for orthogonal changes), but nor do we have another appropriate place.  Three years ago, the right place would have been https://github.com/gravitational/ops, but that repo is a skeleton now.  We could resurrect ops -- but I don't see much point as none of the previous history is relevant to your proposed path forward.

Reply:
> Actions and reusable workflows stored in private repositories cannot be used in public repositories.

The workaround that I've found for this is to use a workflow dispatch event to call a workflow in another repository... although that is definitely messy and probably not the right approach here.

> As such, we may consider a different (new?) repo

I thought about that, but did not originally suggest it because I figured it'd be more work for you and your team. I am not opposed to that if you think it's the right path forward. All that being said, if actions cannot be used from public repos then switching to a private repo (of any form) might make it harder to adopt this change, or may result in a separate private repo down the line. What are your thoughts here?

Reply:
Can we just avoid pushing secrets to this repo?